nextcord/cache.py

```diff
@@ -0,0 +1,102 @@
+from time import time as now
+from typing import Optional, Union
+
+__set_item__ = dict.__setitem__
+__get_item__ = dict.__getitem__
+__get__ = dict.get
+__pop__ = dict.pop
+
+
+class SizedDict(dict):
+    __slots__ = "maxsize"
+
+    def __init__(self, maxsize: Optional[Union[float, int]] = None) -> None:
+        if not maxsize:
+            maxsize = float("inf")
+        else:
+            self.maxsize = maxsize
+
+    def __getitem__(self, key):
+        return __get_item__(self, key)
+
+    def __setitem__(self, key, val) -> None:
+        if len(self) >= self.maxsize:
+            for item in self:
+                del self[item]
+                break
+        __set_item__(self, key, val)
+
+
+class BetterTTLCache(dict):
+    __slots__ = ("maxsize", "ttl", "timelink", "working")
+
+    def __init__(
+        self, maxsize: Optional[int] = None, ttl: Optional[Union[int, float]] = None
+    ) -> None:
+        self.maxsize = maxsize or float("inf")
+        self.ttl = ttl
+        self.timelink = {}
+        self.working = False

```

Comment:
```suggestion
        self.maxsize: float = float(maxsize) or float("inf")
        self.ttl: float = float(ttl)
        self.timelink = {}
        self.working: bool = False

```

maxsize will be inf even if 0 is passed, not sure if this matters.

Reply:
forgot to mention, this is intentional